dev/run-tests.py

```diff
@@ -0,0 +1,417 @@
+                     + sbt_maven_profile_args_base)
+
+
+def is_exe(path):
+    """Check if a given path is an executable file
+    - from: http://stackoverflow.com/a/377028"""
+
+    return os.path.isfile(path) and os.access(path, os.X_OK)
+
+
+def which(program):
+    """Find and return the given program by its absolute path or 'None'
+    - from: http://stackoverflow.com/a/377028"""
+
+    fpath, fname = os.path.split(program)
+
+    if fpath:
+        if is_exe(program):
+            return program
+    else:
+        for path in os.environ.get("PATH").split(os.pathsep):
+            path = path.strip('"')
+            exe_file = os.path.join(path, program)
+            if is_exe(exe_file):
+                return exe_file
+    return None
+
+
+def determine_java_executable():
+    """Will return the *best* path possible for a 'java' executable or `None`"""
+
+    java_home = os.environ.get("JAVA_HOME")
+
+    # check if there is an executable at $JAVA_HOME/bin/java
+    java_exe = which(os.path.join(java_home, "bin/java"))
+    # if the java_exe wasn't set, check for a `java` version on the $PATH
+    return java_exe if java_exe else which("java")
+
+
+def determine_java_version(java_exe):
+    """Given a valid java executable will return its version in named tuple format
+    with accessors '.major', '.minor', '.patch', '.update'"""
+
+    raw_output = subprocess.check_output([java_exe, "-version"], 
+                                         stderr=subprocess.STDOUT)
+    raw_version_str = raw_output.split('\n')[0] # eg 'java version "1.8.0_25"'
+    version_str = raw_version_str.split()[-1].strip('"') # eg '1.8.0_25'
+    version, update = version_str.split('_') # eg ['1.8.0', '25']
+
+    JavaVersion = namedtuple('JavaVersion', 
+                             ['major', 'minor', 'patch', 'update'])
+
+    # map over the values and convert them to integers
+    version_info = map(lambda x: int(x), version.split('.') + [update])
+
+    return JavaVersion(major=version_info[0],
+                       minor=version_info[1],
+                       patch=version_info[2],
+                       update=version_info[3])
+
+
+def multi_starts_with(orig_str, *prefixes):
+    """Takes a string and an abritrary number of prefixes then checks the
+    original string for any of the possible prefixes passed in"""
+
+    for s in prefixes:
+        if orig_str.startswith(s):
+            return True
+    return False
+
+
+def determine_test_suite():
+    """This function current acts to determine if SQL tests need to be run in
+    addition to the core test suite *or* if _only_ SQL tests need to be run
+    as the git logs show that to be the only thing touched. In the future
+    this function will act more generically to help further segregate the
+    test suite runner (hence the function name).
+    @return a set of unique test names"""
+    test_suite = list()
+
+    if AMPLAB_JENKINS:
+        run_cmd(['git', 'fetch', 'origin', 'master:master'])
```

Comment:
This should be updated to diff against the PR's target branch instead of `master`, based on similar changes that @andrewor14 made in https://github.com/apache/spark/pull/6629